store/postgres/src/writable.rs

```diff
@@ -1312,4 +1314,20 @@ impl WritableStoreTrait for WritableStore {
     async fn flush(&self) -> Result<(), StoreError> {
         self.writer.flush().await
     }
+
+    async fn restart(self: Arc<Self>) -> Result<Option<Arc<dyn WritableStoreTrait>>, StoreError> {
+        if self.poisoned() {
+            let logger = self.store.logger.clone();
+            if let Err(e) = self.stop().await {
+                warn!(logger, "Writable had error when stopping, it is safe to ignore this error"; "error" => e.to_string());
+            }
```

Comment:
Is it sufficient to send a stop request here, shouldn't this also join the task handle of the writer, to ensure is finished?

Reply:
I am actually wondering if this is too defensive - if the writer is poisoned, it's because we had an error in the `Queue`, and `start_writer` will return, i.e. shut down, on any error. Maybe I should just remove that code and just leave a comment here?

Reply:
Actually, this would always fail - when a writer has been poisoned, it doesn't allow adding anything more to the queue. I changed this code to just log a warning if the join handle indicates that the background writer is still running. Joining here is a bit tricky since it risks blocking indexing of that subgraph if we really do have a problem with the writer stopping.